src/compositions/cascade/useCascadePanel.ts

```diff
@@ -0,0 +1,53 @@
+import { computed, inject, InjectionKey, MaybeRefOrGetter, provide, Ref, toRef } from 'vue'
+import { CascadePanelId, getInjectedCascadePanels } from '@/compositions'
+
+export const cascadePanelKey: InjectionKey<UseCascadePanel> = Symbol('UseCascadePanel')
+
+export type UseCascadePanel = {
+  isOpen: Ref<boolean>,
+  close: () => void,
+  open: () => void,
+  toggle: () => void,
+}
+
+function getInjectedPanel(): UseCascadePanel {
+  const cascadePanel = inject(cascadePanelKey)
+
+  if (!cascadePanel) {
+    throw new Error('Cascade panel not found. Are you sure the component calling useCascadePanel() exists within a <p-cascade-panel> component context?')
+  }
+
+  return cascadePanel
+}
+
+export function useCascadePanel(id?: MaybeRefOrGetter<CascadePanelId>): UseCascadePanel {
+  if (!id) {
+    return getInjectedPanel()
+  }
+
+  const { closePanelById, openPanelById, togglePanelById, panelIsOpen: globalPanelIsOpen } = getInjectedCascadePanels()
+  const panelId = toRef(id)
+  const isOpen = computed(() => globalPanelIsOpen.value(panelId.value))
+
+  function close(): void {
+    closePanelById(panelId.value)
+  }
+
+  function open(): void {
+    openPanelById(panelId.value)
+  }
+
+  function toggle(): void {
+    togglePanelById(panelId.value)
+  }
+
+  const cascadePanel: UseCascadePanel = {
+    isOpen,
+    toggle,
+    open,
+    close,
+  }
+
+  provide(id, cascadePanel)
```

Comment:
You're providing a `MaybeRefOrGetter` as the key which probably causes some weird things to happen when attempting to inject this. Guessing you meant to provide it with the actual value of the id? I'm surprised typescript isn't complaining since I think the key can only be a string or a symbol. 

Though that kinda contradicts the fact that `MaybeRefOrGetter` implies reactivity. Meaning that if the id changes everything in this composition should change. But `provide` is not reactive. 

Further makes me think this doesn't actually work like you're thinking.

Reply:
Ah nope, that was bad leftover from an earlier version, this wasn't intended to be reactive at all and I ended up poorly mixing the two versions. That should be fixed now, great catch